service/history/conflictResolver.go

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package history
+
+import (
+	"github.com/pborman/uuid"
+	"github.com/uber-common/bark"
+	"github.com/uber/cadence/.gen/go/shared"
+	"github.com/uber/cadence/common"
+	"github.com/uber/cadence/common/persistence"
+)
+
+type (
+	conflictResolver struct {
+		shard              ShardContext
+		context            *workflowExecutionContext
+		historyMgr         persistence.HistoryManager
+		hSerializerFactory persistence.HistorySerializerFactory
+		logger             bark.Logger
+	}
+)
+
+func newConflictResolver(shard ShardContext, context *workflowExecutionContext, historyMgr persistence.HistoryManager,
+	logger bark.Logger) *conflictResolver {
+
+	return &conflictResolver{
+		shard:      shard,
+		context:    context,
+		historyMgr: historyMgr,
+		logger:     logger,
+	}
+}
+
+func (r *conflictResolver) reset(replayEventID int64) (*mutableStateBuilder, error) {
```

Comment:
just want to confirm, after the conflict resolution, the transfer / timer task will be created and notified?

Reply:
Yes.